File: server-java/src/main/java/com/zhiyu/health/entity/prescription/Medication.java
package com.zhiyu.health.entity.prescription;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Getter;
import lombok.Setter;

/**
 * 标准药品目录（票 88，ADR-0035）：全局目录只承载药品身份与知识字段，
 * 价格/库存/在售语义已收敛到各院区药房的 pharmacy_medications。
 */
@Getter
@Setter
@TableName("medications")
public class Medication {
    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;
    private String genericName;
    private String specification;
    private String instructions;
    // 票 76（ADR-0032）：处方药 TRUE 须凭已审核处方，OTC FALSE 可直接下单；DEFAULT TRUE 偏安全。
    private Boolean isPrescription;
}

File: admin/src/services/medication.ts
import { request } from '@umijs/max';

// 票 88（ADR-0035）：medications 收敛为全平台标准药品目录，
// 不再承载价格/库存/启用状态（这些语义已下沉到院区药房药品 pharmacy_medications）。
export interface Medication {
  id: number;
  name: string;
  generic_name: string;
  specification: string;
  instructions: string;
  is_prescription: boolean;
}

export interface MedicationInput {
  is_prescription: boolean;
}

export const listMedications = () => request<Medication[]>('/api/b/medications');

export const updateMedication = (id: number, body: MedicationInput) =>
  request<Medication>(`/api/b/medications/${id}`, { method: 'PUT', data: body });
